core-drawer-panel.html

```diff
@@ -75,344 +75,425 @@
-      this.hasWillChange = 'willChange' in this.style;
-    },
-
-    domReady: function() {
-      // to avoid transition at the beginning e.g. page loads
-      // NOTE: domReady is already raf delayed and delaying another frame
-      // ensures a layout has occurred.
-      this.async(function() {
-        this.transition = true;
-      });
-    },
-
-    /**
-     * Toggles the panel open and closed.
-     *
-     * @method togglePanel
-     */
-    togglePanel: function() {
-      this.selected = this.isMainSelected() ? 'drawer' : 'main';
-    },
-
-    /**
-     * Opens the drawer.
-     *
-     * @method openDrawer
-     */
-    openDrawer: function() {
-      this.selected = 'drawer';
-    },
-
-    /**
-     * Closes the drawer.
-     *
-     * @method closeDrawer
-     */
-    closeDrawer: function() {
-      this.selected = 'main';
-    },
-
-    queryMatchesChanged: function() {
-      this.narrow = this.queryMatches || this.forceNarrow;
-      if (this.narrow) {
-        this.selected = this.defaultSelected;
-      }
-      this.setAttribute('touch-action', this.swipeAllowed() ? 'pan-y' : '');
-      this.fire('core-responsive-change', {narrow: this.narrow});
-    },
-    
-    forceNarrowChanged: function() {
-      this.queryMatchesChanged();
-    },
-
-    swipeAllowed: function() {
-      return this.narrow && !this.disableSwipe;
-    },
-    
-    isMainSelected: function() {
-      return this.selected === 'main';
-    },
-
-    startEdgePeek: function() {
-      this.width = this.$.drawer.offsetWidth;
-      this.moveDrawer(this.translateXForDeltaX(this.rightDrawer ?
-          -this.edgeSwipeSensitivity : this.edgeSwipeSensitivity));
-      this.peeking = true;
-    },
-
-    stopEdgePeak: function() {
-      if (this.peeking) {
-        this.peeking = false;
-        this.moveDrawer(null);
-      }
-    },
+      closeDrawer: function() {
+        this.selected = 'main';
+      },
 
-    downHandler: function(e) {
-      if (!this.dragging && this.isMainSelected() && this.isEdgeTouch(e)) {
-        this.startEdgePeek();
-      }
-    },
-
-    upHandler: function(e) {
-      this.stopEdgePeak();
-    },
-    
-    tapHandler: function(e) {
-      if (e.target && this.toggleAttribute && 
-          e.target.hasAttribute(this.toggleAttribute)) {
-        this.togglePanel();
-      }
-    },
+      onQueryMatchesChanged: function(event) {
+        this.narrow = event.detail.value || this.forceNarrow;
 
-    isEdgeTouch: function(e) {
-      return !this.disableEdgeSwipe && this.swipeAllowed() &&
-        (this.rightDrawer ?
-          e.pageX >= this.offsetWidth - this.edgeSwipeSensitivity :
-          e.pageX <= this.edgeSwipeSensitivity);
-    },
+        if (this.narrow) {
+          this.selected = this.defaultSelected;
+        }
 
-    trackStart : function(e) {
-      if (this.swipeAllowed()) {
-        this.dragging = true;
+        this.setAttribute('touch-action', this.swipeAllowed() ? 'pan-y' : '');
```

Comment:
Even though this isn't computed, perhaps we can write an accessor for the value?